### PR TITLE
Fix: Removed duplicate mobile folder

### PR DESCRIPTION
**Pull request by gregorysa**
# Description
- Removed extra (tabs) folder
- Moved the event and saved tab into the (app)/(tabs) folder
- Removed the hardcoded guardian ID from the saved tab
- Error handling for no guardian ID

# How Has This Been Tested?
Please describe the tests that you manually ran to verify your changes (beyond any unit/integration tests written and ran).

## Screenshots
Please provide screenshots of manual testing (Scalar, frontend pages, etc.)

# Checklist

## General
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas

## Backend Changes (if applicable)
- [ ] I have written unit tests for my code
- [ ] I have written integration tests where appropriate
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have run make api-validate and make api-gen to keep the openAPI spec up to date
- [ ] I have tested error handling and edge cases
- [ ] Database migrations are included (if applicable)

## Frontend Changes (if applicable)
- [ ] I have tested in multiple browsers (Chrome, Firefox, Safari)
- [ ] Mobile responsive design verified (if applicable)
- [ ] I have run make generate-api to keep the API hooks up to date
- [ ] **For mobile features:** Tested on both iOS and Android devices
- [ ] **For mobile features:** Screenshots from both iOS and Android included below
- [ ] No console errors or warnings
- [ ] Loading states and error states handled

### iOS
[Add iOS screenshots here]

### Android
[Add Android screenshots here]

### CODE REVIEW

File: frontend/apps/mobile/app/(app)/(tabs)/index.tsx [hunk 1]
```diff
@@ -98,12 +82,16 @@ function FilterChips({
...
+  const router = useRouter();
   const ageLabel = event.event.age_range_min != null
     ? `${event.event.age_range_min}${event.event.age_range_max != null ? `–${event.event.age_range_max}` : ""}+`
     : null;
 
   return (
-    <View style={{ marginHorizontal: 20, borderRadius: 24, overflow: "hidden", height: 180, backgroundColor: "#111" }}>
+    <Pressable
+      onPress={() => router.push(`/event/${event.id}`)}
+      style={{ marginHorizontal: 20, borderRadius: 24, overflow: "hidden", height: 180, backgroundColor: AppColors.primaryText }}
```
Comment (josh-torre) [inline frontend/apps/mobile/app/(app)/(tabs)/index.tsx]: use the classname for static styling

File: frontend/apps/mobile/components/BookmarkButton.tsx [hunk 2]
```diff
@@ -45,28 +47,34 @@ export function BookmarkButton({ occurrenceId }: { occurrenceId: string }) {
   const createSaved = useCreateSaved(
     optimisticOptions((items) => [
       ...items,
-      { id: "optimistic", guardian_id: GUARDIAN_ID, event_occurrence_id: occurrenceId, created_at: "", updated_at: "" },
+      { id: "optimistic", guardian_id: guardianId!, event: { id: eventId }, created_at: "", updated_at: "" } as Saved,
```
Comment (josh-torre) [inline frontend/apps/mobile/components/BookmarkButton.tsx]: Nit: pref if we define this outside of the list so that we can drop the cast
